packages/common/aws.js

```diff
@@ -500,6 +500,52 @@ exports.sendSQSMessage = (queueUrl, message) => {
   }).promise();
 };
 
+/**
+ * Receives SQS messages from a given queue. The number of messages received
+ * can be set and the timeout is also adjustable.
+ *
+ * @param {string} queueUrl - url of the SQS queue
+ * @param {integer} numOfMessages - number of messages to read from the queue
+ * @param {integer} timeout - number of seconds it takes for a message to timeout
+ * @returns {Promise.<Array>} an array of messages
+ */
+exports.receiveSQSMessages = async (queueUrl, numOfMessages = 1, timeout = 30) => {
+  const params = {
+    QueueUrl: queueUrl,
+    AttributeNames: ['All'],
+    VisibilityTimeout: timeout,
+    MaxNumberOfMessages: numOfMessages
+  };
+
+  const messages = await exports.sqs().receiveMessage(params).promise();
+
+  // convert body from string to js object
+  if (Object.prototype.hasOwnProperty.call(messages, 'Messages')) {
+    messages.Messages.forEach((mes) => {
+      mes.Body = JSON.parse(mes.Body); // eslint-disable-line no-param-reassign
+    });
+
+    return messages.Messages;
+  }
+  return [];
```

Comment:
I think this function just returns an array of messages (or the empty array) (e.g. not `{Promise.<Array>}` as documentation above indicates

Reply:
async/await function always return a promise.

Reply:
You're right! I learned something.